src/Microsoft.Repl.Tests/Mocks/MockConsoleManager.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Threading;
+using Microsoft.Repl.ConsoleHandling;
+
+namespace Microsoft.Repl.Tests.Mocks
+{
+    internal class MockConsoleManager : IConsoleManager
+    {
+        private CancellationTokenSource _cancellationTokenSource;
+        private ConsoleKeyInfo _consoleKeyInfo;
+
+        public MockConsoleManager(ConsoleKeyInfo consoleKeyInfo, CancellationTokenSource cancellationTokenSource)
+        {
+            _cancellationTokenSource = cancellationTokenSource;
+            _consoleKeyInfo = consoleKeyInfo;
+        }
+
+        public Point Caret => throw new NotImplementedException();
+
+        public Point CommandStart => throw new NotImplementedException();
+
+        public int CaretPosition { get; set; }
+
+        public IWritable Error => new MockWritable();
+
+        public bool IsKeyAvailable => throw new NotImplementedException();
+
+        public void Clear()
+        {
+        }
+
+        public void MoveCaret(int position)
```

Comment:
Can you rename this to `offset` instead of `position`, including in the IConsoleManager interface.  Using `position` is a confusing name because it sounds like "MoveCaret To Position", not "MoveCaret By Offset" (which is what both implementations actually do).